dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/deduplication/DeduplicationHelper.java

```diff
@@ -59,7 +60,40 @@
 
     public MergeObject generateMergeObject( TrackedEntityInstance original, TrackedEntityInstance duplicate )
     {
-        return null;
+        if ( duplicate == null )
```

Comment:
We can assume original and duplicate are never null, due to the flow; We check if both exist in the controller. However, assume we dont know that, I would make sure both are not null :)

Reply:
check for null is now removed